flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/global/GlobalCommitterOperator.java

```diff
@@ -103,18 +115,20 @@ public void snapshotState(StateSnapshotContext context) throws Exception {
         if (committables.size() > 0) {
             committablesPerCheckpoint.put(
                     context.getCheckpointId(),
-                    globalCommitter.combine(context.getCheckpointId(), committables));
+                    committer.combine(context.getCheckpointId(), committables));
         }
         streamingCommitterState.update(new ArrayList<>(committablesPerCheckpoint.values()));
     }
 
     @Override
     public void endInput() throws Exception {
-        List<CommT> allCommittables = pollCommittables();
-        if (!allCommittables.isEmpty()) {
-            globalCommitter.commit(
-                    Collections.singletonList(
-                            globalCommitter.combine(Long.MAX_VALUE, allCommittables)));
+        if (isBounded) {
+            List<CommT> allCommittables = pollCommittables();
+            if (!allCommittables.isEmpty()) {
+                committer.commit(
+                        Collections.singletonList(
+                                committer.combine(Long.MAX_VALUE, allCommittables)));
+            }
```

Comment:
Why do we need to check this? Why not commit directly for all cases? If the source is not bounded will `endInput()` even be called?

Reply:
You can see the document in `BoundedOneInput`.

> It is not safe to use this method to commit any transactions or other side effects! You can use this method to flush any buffered data that can later on be committed e.g. in a `{@link StreamOperator#notifyCheckpointComplete(long)}`.

Reply:
It is better to name this field to `isBatchMode`.

Reply:
Not exactly. We should also consider the case when checkpointing is not enabled for a streaming job. Actually FLINK-25941 also states this (and also affects the compilation of this module). Please take a look.

Reply:
We can discuss this later, I'll remove this commit in the PR.